src/main/java/com/insightfullogic/honest_profiler/core/collector/LogCollector.java

```diff
@@ -21,27 +21,31 @@
  **/
 package com.insightfullogic.honest_profiler.core.collector;
 
+import static java.util.Comparator.comparing;
```

Comment:
Sorry about that. The meaningful bit amounts to :

136c140,143
<         metaByThreadId.put(newThreadMeta.getThreadId(), newThreadMeta);
---
>         metaByThreadId.merge(
>             newThreadMeta.getThreadId(),
>             newThreadMeta,
>             (oldMeta, newMeta) -> oldMeta.update(newMeta));